src/dev-app/performance/performance-demo.html

```diff
@@ -0,0 +1,80 @@
+<div>
+  <mat-form-field appearance="outline" style="margin-right: 32px">
+    <mat-label>Sample size</mat-label>
+    <mat-select [(value)]="sampleSize" [disabled]="isRunningBenchmark">
+      <mat-option [value]="1">1</mat-option>
+      <mat-option [value]="10">10</mat-option>
+      <mat-option [value]="100">100</mat-option>
+      <mat-option [value]="1000">1000</mat-option>
+    </mat-select>
+  </mat-form-field>
+  <mat-form-field appearance="outline">
+    <mat-label>Component count</mat-label>
+    <mat-select [(value)]="componentCount" [disabled]="isRunningBenchmark">
+      <mat-option [value]="1">1</mat-option>
+      <mat-option [value]="10">10</mat-option>
+      <mat-option [value]="100">100</mat-option>
+      <mat-option [value]="1000">1000</mat-option>
+    </mat-select>
+  </mat-form-field>
+</div>
+
+<div class="demo-table-and-paginator-container">
+  <div class="demo-table-container">
+    <table mat-table [dataSource]="dataSource" class="mat-elevation-z8">
+      <ng-container matColumnDef="index">
+        <th mat-header-cell *matHeaderCellDef>No.</th>
+        <td mat-cell *matCellDef="let item">{{ item.index }}</td>
+        <td mat-footer-cell *matFooterCellDef>
+          <ng-container *ngIf="allSamples.length">
+            <b>Total render time</b>
+          </ng-container>
+          <ng-container *ngIf="!allSamples.length"> No data yet </ng-container>
+        </td>
+      </ng-container>
+
+      <ng-container matColumnDef="time">
+        <th mat-header-cell *matHeaderCellDef>Time</th>
+        <td mat-cell *matCellDef="let item">{{ item.time }}</td>
+        <td mat-footer-cell *matFooterCellDef>
+          <b>{{ totalRenderTime }}</b>
+        </td>
+      </ng-container>
+
+      <tr mat-header-row *matHeaderRowDef="displayedColumns; sticky: true"></tr>
+      <tr mat-row *matRowDef="let row; columns: displayedColumns"></tr>
+      <tr mat-footer-row *matFooterRowDef="displayedColumns; sticky: true"></tr>
+    </table>
+  </div>
+  <mat-divider></mat-divider>
+  <mat-paginator
+    [pageSizeOptions]="[10, 50, 100]"
+    [pageSize]="100"
+    showFirstLastButtons
+    [disabled]="isRunningBenchmark"
+  ></mat-paginator>
+</div>
+
+<button
+  color="accent"
+  mat-button
+  (click)="clearMetrics()"
+  [disabled]="isRunningBenchmark"
+  style="margin-right: 32px"
+  *ngIf="allSamples.length"
+>
+  Clear Metrics
+</button>
+
+<button color="primary" mat-raised-button (click)="runBenchmark()" [disabled]="isRunningBenchmark">
+  Run Benchmark
+</button>
+
+<ng-container *ngIf="show">
+  <ng-container *ngFor="let _ of [].constructor(componentCount)">
```

Comment:
I think that this is going to cause a bunch of unnecessary change detections, because `[].constructor` will produce a new array any time it is checked.